module/VuFindApi/src/VuFindApi/Formatter/RecordFormatter.php

```diff
@@ -0,0 +1,262 @@
+ * @link     https://vufind.org/wiki/development:plugins:controllers Wiki
+ */
+class RecordFormatter extends BaseFormatter
+{
+    /**
+     * Record field definitions
+     *
+     * @var array
+     */
+    protected $recordFields;
+
+    /**
+     * View helper plugin manager
+     *
+     * @var HelperPluginManager
+     */
+    protected $helperManager;
+
+    /**
+     * Constructor
+     *
+     * @param array               $recordFields  Record field definitions
+     * @param HelperPluginManager $helperManager View helper plugin manager
+     */
+    public function __construct($recordFields, HelperPluginManager $helperManager
+    ) {
+        $this->recordFields = $recordFields;
+        $this->helperManager = $helperManager;
+    }
+
+    /**
+     * Get dedup IDs
+     *
+     * @param \VuFind\RecordDriver\AbstractBase $record Record driver
+     *
+     * @return array|null
+     */
+    protected function getDedupIds($record)
+    {
+        if (!($dedupData = $record->tryMethod('getDedupData'))) {
+            return null;
+        }
+        $result = [];
+        foreach ($dedupData as $item) {
+            $result[] = $item['id'];
+        }
+        return $result ? $result : null;
+    }
+
+    /**
+     * Get full record for a record as XML
+     *
+     * @param \VuFind\RecordDriver\AbstractBase $record Record driver
+     *
+     * @return string|null
+     */
+    protected function getFullRecord($record)
+    {
+        if ($xml = $record->tryMethod('getFilteredXML')) {
+            return $xml;
+        }
+        $rawData = $record->tryMethod('getRawData');
+        return isset($rawData['fullrecord']) ? $rawData['fullrecord'] : null;
+    }
+
+    /**
+     * Get raw data for a record as an array
+     *
+     * @param \VuFind\RecordDriver\AbstractBase $record Record driver
+     *
+     * @return array
+     */
+    protected function getRawData($record)
+    {
+        $rawData = $record->tryMethod('getRawData');
+
+        // Leave out spelling data
+        unset($rawData['spelling']);
+
+        return $rawData;
+    }
+
+    /**
+     * Get (relative) link to record page
+     *
+     * @param \VuFind\RecordDriver\AbstractBase $record Record driver
+     *
+     * @return string
+     */
+    protected function getRecordPage($record)
+    {
+        $urlHelper = $this->helperManager->get('recordLink');
+        return $urlHelper->getUrl($record);
+    }
+
+    /**
+     * Get source
+     *
+     * @param \VuFind\RecordDriver\AbstractBase $record Record driver
+     *
+     * @return array|null
+     */
+    protected function getSource($record)
+    {
+        if ($sources = $record->tryMethod('getSource')) {
+            $result = [];
+            $translator = $this->helperManager->get('translate');
+            foreach ($sources as $source) {
+                $result[] = [
+                    'value' => $source,
+                    'translated' => $translator->translate("source_$source")
```

Comment:
That's indeed local to us and predates the availability of text domains. I think it's safe to remove from the basic API for now since it doesn't return anything useful in the default Solr drivers.